holopy/scattering/tests/test_lens.py

```diff
@@ -0,0 +1,213 @@
+import unittest
+
+import numpy as np
+from numpy.testing import assert_allclose, assert_equal
+
+from holopy.core import detector_points, update_metadata, detector_grid
+from holopy.scattering import calc_holo
+from holopy.scattering.theory import Mie, MieLens
+from holopy.scattering.theory.lens import Lens
+from holopy.scattering.theory.mielensfunctions import MieLensCalculator
+import holopy.scattering.tests.common as test_common
+
+LENS_ANGLE = 1.
+QLIM_TOL = {'atol': 1e-2, 'rtol': 1e-2}
+LENSMIE = Lens(lens_angle=LENS_ANGLE, theory=Mie(False, False))
+
+class TestLens(unittest.TestCase):
+    def test_can_handle(self):
+        theory = LENSMIE
+        self.assertTrue(theory._can_handle(test_common.sphere))
+
+    def test_theta_quad_pts_min(self):
+        assert_allclose(np.min(LENSMIE._theta_pts), 0., **QLIM_TOL)
+
+    def test_theta_quad_pts_max(self):
+        assert_allclose(np.max(LENSMIE._theta_pts), LENS_ANGLE, **QLIM_TOL)
+
+    def test_theta_quad_pts_num(self):
+        assert_equal(len(LENSMIE._theta_pts), len(LENSMIE._theta_wts))
+
+    def test_phi_quad_pts_min(self):
+        assert_allclose(np.min(LENSMIE._phi_pts), 0., **QLIM_TOL)
+
+    def test_phi_quad_pts_max(self):
+        assert_allclose(np.max(LENSMIE._phi_pts), 2*np.pi, **QLIM_TOL)
+
+    def test_phi_quad_pts_num(self):
+        assert_equal(len(LENSMIE._phi_pts), len(LENSMIE._phi_wts))
+
+    def test_integrate_over_theta_with_quad_points(self):
+        pts = LENSMIE._theta_pts
+        wts = LENSMIE._theta_wts
+        func = np.cos
+        integral = np.sum(func(pts) * wts)
+        expected_val = np.sin(LENSMIE.lens_angle) # analytic result
+        assert_allclose(integral, expected_val)
+
+    def test_integrate_over_phi_with_quad_points(self):
+        pts = LENSMIE._phi_pts
+        wts = LENSMIE._phi_wts
+        func = lambda x: np.cos(np.pi * x)
+        integral = np.sum(func(pts) * wts)
+        expected_val = np.sin(2 * np.pi ** 2) /  np.pi # analytic result
```

Comment:
Wait I'm confused. The integral of cos(pi*x) from 0 to 2pi is sin(2*pi^2) / pi? Shouldn't it be zero?

If it is zero, that may be a sign that the default tolerances in `assert_allclose` are too soft; it might be good to tighten them.

If I'm just confused, ignore me :smile: